src/main/java/sg/ncl/MainController.java

```diff
@@ -1918,6 +1948,94 @@ public String validateExperiment(
         return "redirect:/experiments";
     }
 
+    @RequestMapping(value = "/experiments/save_image/{teamId}/{expId}/{nodeId}", method = RequestMethod.GET)
+    public String saveExperimentImage(@PathVariable String teamId, @PathVariable String expId, @PathVariable String nodeId, Model model) {
+        Map<String, Map<String, String>> singleNodeInfoMap = new HashMap<>();
+        Image saveImageForm = new Image();
+
+        String teamName = invokeAndExtractTeamInfo(teamId).getName();
+        Realization realization = invokeAndExtractRealization(teamName, Long.parseLong(expId));
+
+        // experiment may have many nodes
+        // extract just the particular node details to display
+        for (Map.Entry<String, Map<String, String>> nodesInfo : realization.getNodesInfoMap().entrySet()) {
+            String nodeName = nodesInfo.getKey();
+            Map<String, String> singleNodeDetailsMap = nodesInfo.getValue();
+            if (singleNodeDetailsMap.get("nodeId").equals(nodeId)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "nodeId" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)